### PR TITLE
TDD testing

### PR DESCRIPTION
**Pull request by theBwoyProgrammer**
## Test Driven Development
- Created a class called `Solver`
- Created a method called `factorial` that takes one argument, an integer `N`, and returns the factorial for that number
- Created a method called `reverse` that takes one argument, a string word, and returns a word reversed
- Created a method called `fizzbuzz` that takes one argument, an integer `N`, and returns a string;
1. When N is divisible by 3, return "fizz".
2. When N is divisible by 5, return "buzz".
3. When N is divisible by 3 and 5, return "fizzbuzz".
4. Any other case, return N as a string (e.g. say N is 7 then return "7").

### CODE REVIEW

File: spec/solver.spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+require_relative '../solver'
+
+RSpec.describe ' Solver' do
+  describe '#factorial' do
+    it 'should return factorial of a number' do
+      res = Solver.new
+      expect(res).to receive(:factorial).with(5).and_return(120)
+      expect(res.factorial(5)).to eq(120)
+    end
+  end
```
Comment (DammyShittu) [inline spec/solver.spec.rb]: - [x] Kindly include a scenario for when the number is negative, in which case it should raise an exception.

- [x] Include test for when the number is 0, the result should be 1

File: spec/solver.spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  describe '#fizzbuzz' do
+    it 'should return fizz if divisible by 3' do
+      res = Solver.new
+      expect(res).to receive(:fizzbuzz).with(3).and_return('fizz')
+      expect(res.fizzbuzz(3)).to eq('fizz')
+    end
+  end
```
Comment (DammyShittu) [inline spec/solver.spec.rb]: - [x] Include more tests for the following scenarios. When a number is

  - divisible by ONLY 5 it should return 'buzz'
  - divisible by BOTH 3 and 5 it should return 'fizzbuzz'
  - not divisible by either 3 and 5, it should return a string version of that number

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @techwolf404 @TuyishimireEric,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/solver.spec.rb (hunk 1 above)
Comment (theBwoyProgrammer) [reply spec/solver.spec.rb]: Ok, let us fix that. Thank you

File: spec/solver.spec.rb (hunk 2 above)
Comment (theBwoyProgrammer) [reply spec/solver.spec.rb]: done

**Review by mikethreels — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job following the instructions of the previous CR.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**